services/device-registry-base/src/main/java/org/eclipse/hono/service/management/BaseDto.java

```diff
@@ -22,11 +22,14 @@
 import com.fasterxml.jackson.annotation.JsonInclude;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
+import io.quarkus.runtime.annotations.RegisterForReflection;
+
 /**
  * The base class for implementing a DTO (Data Transfer Object) to store data in registry implementations.
  *
  * @param <T> The type of the data object
  */
+@RegisterForReflection
```

Comment:
We could, but using the annotation makes it much more explicit than using the (somewhat hidden) reflection config file. Actually, using the annotation is the approach recommended in the Quarkus documentation. Using the config file is helpful for marking third party classes for reflection, though.

Reply:
I also think that the annotation is more explicit. IMHO we should use the same approach in other modules too instead of registering internal classes for reflection using a config file.

Reply:
I agree, but we should do that in a separate PR, ok?